modules/standard/helpbot/kos_list_controller.py

```diff
@@ -0,0 +1,75 @@
+from core.chat_blob import ChatBlob
+from core.command_param_types import Character, Const, Options, Any
+from core.decorators import instance, command
+from core.sender_obj import SenderObj
+from core.standard_message import StandardMessage
+import time
+
+@instance()
+class KosListController:
+    def __init__(self):
+        pass
+
+    def inject(self, registry):
+        self.bot = registry.get_instance("bot")
+        self.buddy_service = registry.get_instance("buddy_service")
+        self.db = registry.get_instance("db")
+        self.pork_service = registry.get_instance("pork_service")
+
+    def start(self):
+        self.db.exec("CREATE TABLE IF NOT EXISTS kos_list (target_id INT PRIMARY KEY, submitter_id INT NOT NULL,"
+                     " submitted_at INT NOT NULL, reason VARCHAR(2000) NOT NULL DEFAULT '')")
+    
+    @command(command="kos", params=[], access_level="member", description="Shows the kill-on-sight list")
+    def kos_list_show_cmd(self, request):
+        entries = self.db.query("SELECT kos_list.target_id AS target_id, p1.name AS target_name, "
+                                "p2.name AS submitter_name, kos_list.submitted_at AS submitted_at, kos_list.reason AS reason "
+                                "FROM kos_list "
+                                "LEFT JOIN player p1 ON kos_list.target_id = p1.char_id "
+                                "LEFT JOIN player p2 ON kos_list.submitter_id = p2.char_id "
+                                "ORDER BY kos_list.submitted_at DESC")
+        count = len(entries)
+        
+        if count == 0:
+            return "The kill-on-sight list is empty"
+        
+        blob = ""
+        for entry in entries:
+            target_info = self.pork_service.get_character_info(entry.target_name)
```

Comment:
this will trigger a call to pork for each name on the list, which could be slow once there are more than just a few names.  instead, we should get the info from the player table in the sql

Reply:
hm, then i will fetch the PoRK data once in the add command, because else I have quite empty rows in the `player` table 
![image](https://github.com/Budabot/Tyrbot/assets/31340422/42de0c64-ad4a-4053-8ec0-5d1cb19d0c19)
(take a look at `Captestslav2`, its in there from a test before i was fetching PoRK info)

Reply:
perfect, good idea